localstack-core/localstack/services/sqs/models.py

```diff
@@ -606,9 +607,17 @@ def requeue_inflight_messages(self):
                     standard_message,
                     self.arn,
                 )
-                self.inflight.remove(standard_message)
+                del self.inflight[standard_message]
                 self._put_message(standard_message)
 
+    def add_inflight_message(self, message: SqsMessage):
+        """
+        We are simulating an ordered set with a dict. When a value is added, it is added as key to the dict, which
+        is all we need. Hence all "values" in this ordered set are None
+        :param message: The message to put in flight
+        """
+        self.inflight[message] = None
```

Comment:
To avoid confusion when adding to the ordered set, all messages added should use this method (simply because `self.inflight[message] = None` is very vague and non-descript on its own. It is still not pretty, but it works